### PR TITLE
Portfolio optimization

### PR DESCRIPTION
**Pull request by Qinlu-HD**
add the full update operator of 3s operator, add unit test for the full update of 3s operator.

### CODE REVIEW

File: include/operators.h [hunk 1]
```diff
@@ -1443,8 +1443,54 @@ class portfolio_3s : public OperatorInterface {
...
+    
+    Vector y(len, 0.);
+    y = (*v_in);
+    if (w1<=0 && w2>=0) {
+      y = (*v_in);
+    }else if (w2<=0 && w3>=0) {
+      add(y, a2, -w2);
+    }else if (w3<=0 && w4>=0) {
+      add(y, tilde_a1, -w1);
+      add(y, tilde_a2, -w2);
```
Comment (ZhiminPeng) [inline include/operators.h]: tild_a1 and tild_a2 are not defined.

File: test/operators_unittest.cc [hunk 2]
```diff
@@ -2159,13 +2159,13 @@ TEST(PortfolioTest, VectorOperator) {
   int n = 2;
   Vector v(n, 0.);
   double step_size = 1.;
-  int m = 2;
   
   // test for Matrix
   Matrix Q(m, n, 1.);
```
Comment (ZhiminPeng) [inline test/operators_unittest.cc]: m is not defined.

**Comment by ZhiminPeng** [comment]
Before committing the code, please make sure that all of the unit tests are passed.

**Comment by Qinlu-HD** [comment]
add 3s splitting schemes

**Comment by Qinlu-HD** [comment]
fixed some bugs in 3s operator, fixed some bugs in unittest for 3s operator, and add 3s splitting schemes

**Pull request closed without merging**